src/Razor/test/Microsoft.VisualStudio.Razor.IntegrationTests/AbstractRazorEditorTest.cs

```diff
@@ -66,6 +68,37 @@ private static void EnsureLSPEditorEnabled()
             Assert.AreEqual(false, useLegacyEditor, "Expected the Legacy Razor Editor to be disabled, but it was enabled");
         }
 
+        private static void EnsureMEFCompositionSuccessForRazor()
+        {
+            var hiveDirectories = VisualStudioLogging.GetHiveDirectories();
+            if (hiveDirectories.Count() != 1)
+            {
+                throw new ArgumentOutOfRangeException("Should only have one hive");
+            }
+
+            var hiveDirectory = hiveDirectories.Single();
```

Comment:
The double enumeration of hiveDirectories here disgusts me, but its only test code, and all crimes are forgiven in test code.